### PR TITLE
Handle special domains better

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @MariagraziaAlastra 
**CC:** @dharb 

## Description:

Just tightening up how we determine what's a special page and what isn't.

(Note, there's an outstanding bug where the popup might not update if you go from a non-special page to a special page. This PR doesn't fix that one.)

## Steps to test this PR:

Open a bunch of special pages on Chrome, Firefox and Vivaldi.
None of them should be showing a grade, and the description should be sane.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/chrome-wrapper.es6.js [hunk 1]
```diff
@@ -21,6 +21,10 @@ let getFromStorage = (key, cb) => {
     })
 }
 
+let getExtensionId = () => {
+    return chrome.runtime.id
+}
```
Comment (andrey-p) [inline shared/js/background/chrome-wrapper.es6.js]: Moved it to chrome-wrapper so I can stub it more easily.

File: shared/js/background/classes/site.es6.js [hunk 2]
```diff
@@ -43,8 +44,8 @@ class Site {
             this.grade.setHttps(true, true)
         }
 
-        // set isSpecialDomain when the site is created
-        this.isSpecialDomain = this.specialDomain()
+        // set specialDomainName when the site is created
+        this.specialDomainName = this.getSpecialDomain()
```
Comment (andrey-p) [inline shared/js/background/classes/site.es6.js]: Sorry, couldn't resist renaming these. Variables that pretend to be booleans and method names that pretend to not be method names are a pet peeve of mine.

File: shared/js/background/classes/site.es6.js [hunk 3]
```diff
@@ -103,24 +104,56 @@ class Site {
...
-        if (this.domain === 'extensions') { return 'extensions' }
+    getSpecialDomain () {
+        const extensionId = browserWrapper.getExtensionId()
+        const url = this.url
+        let domain = this.domain
 
-        if (window.chrome && this.domain === chrome.runtime.id) { return 'options' }
+        if (url === '') {
+            return 'new tab'
+        }
```
Comment (andrey-p) [inline shared/js/background/classes/site.es6.js]: In the previous version, we only did that if the browser is FF. I can't imagine a case in which the URL will be empty but we wouldn't want to consider it a new tab.

File: shared/js/background/utils.es6.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ const tldjs = require('tldjs')
 const entityMap = require('../../data/tracker_lists/entityMap')
 
 function extractHostFromURL (url, shouldKeepWWW) {
-    if (!url) return
+    if (!url) return ''
```
Comment (andrey-p) [inline shared/js/background/utils.es6.js]: This was erroring out in the empty URL case. Not sure if it was ever hit in production, but I think it's worth maintaining consistency between different outputs.

(E.g. if a caller expects a function to return a string, always try and return a string.)

**Comment by MariagraziaAlastra** [comment]
@andrey-p this looks good, only thing is on Vivaldi the message for the new tab is "extension page" as opposed to "new tab". I think it's pretty minor though, and it improves functionality, so LGTM 👍

**Pull request merged**